### PR TITLE
Add a harder testcase for logit check

### PR DESCRIPTION
**Pull request by JamesDeng42**
# Description

This introduces a more complex test case to compare logits between two models during the forward pass. Previous tests were too weak and could potentially mask subtle differences. This stricter test ensures any discrepancies are reliably exposed.

# Tests
Tested locally.

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [X] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [X] I have necessary comments in my code, particularly in hard-to-understand areas.
- [X] I have run end-to-end tests tests and provided workload links above if applicable.
- [X] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

**Comment by vlad-karp** [comment]
@JamesDeng42 , what about a sequence of length which is close to the limit of some popular default model ?

**Comment by JamesDeng42** [comment]
> @JamesDeng42 , what about a sequence of length which is close to the limit of some popular default model ?

**Comment by JamesDeng42** [comment]
> @JamesDeng42 , what about a sequence of length which is close to the limit of some popular default model ?

I think we need a better hard case especially for DS v3 and Qwen3 which are in our target. Let me think about a better way to do this. Maybe a hard case for 8B model and even harder case for Qwen3-480B-MoE if an additional flag is turn on. what do you think?

**Comment by vlad-karp** [comment]
> > @JamesDeng42 , what about a sequence of length which is close to the limit of some popular default model ?
> 
> I think we need a better hard case especially for DS v3 and Qwen3 which are in our target. Let me think about a better way to do this. Maybe a hard case for 8B model and even harder case for Qwen3-480B-MoE if an additional flag is turn on. what do you think?
The test is using a specific model, let's use and example of a maximum (or even exceeding) length of that model
